01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -1,18 +1,33 @@
 class MyQueue
   attr_accessor :head
   attr_accessor :tail
+  attr_accessor :queue_length
 
   def initialize
     @queue = Array.new
     @head = @queue[0]
+    @queue_length = @queue.length
   end
 
   def enqueue(element)
+    @queue[@queue.length] = element
+    @head = @queue[0]
```

Comment:
enqueueing doesn't involve setting a new head unless it's the very first element